### PR TITLE
refactor(UserFlags): update flag names

### PR DESCRIPTION
**Pull request by almeidx**
**Please describe the changes this PR makes and why it should be merged:**
~~I changed `HypeSquad` to `Hypesquad` since that's what Discord calls it (`HYPESQUAD` and not `HYPE_SQUAD`), let me know if you want me to change it back to HypeSquad.~~

**Reference Discord API Docs PRs or commits:**
- https://github.com/discord/discord-api-docs/pull/4036

### CODE REVIEW

**Comment by almeidx** [comment]
Should I also add a comment to each key with the flags description?

**Comment by vladfrangu** [comment]
It's funky with HypeSquad.. the name is one word, so it should be `Hypesquad` but their description says `HypeSquad`... @discordjs/the-big-4 thoughts? I prefer `HypeSquad` myself, but I'd love your thoughts too

File: payloads/v8/user.ts [hunk 1]
```diff
@@ -86,19 +86,19 @@ export interface APIUser {
...
+	HypesquadOnlineHouse2 = 1 << 7,
+	HypesquadOnlineHouse3 = 1 << 8,
+	PremiumEarlySupporter = 1 << 9,
+	TeamPseudoUser = 1 << 10,
 	BugHunterLevel2 = 1 << 14,
 	VerifiedBot = 1 << 16,
-	EarlyVerifiedBotDeveloper = 1 << 17,
-	DiscordCertifiedModerator = 1 << 18,
+	VerifiedDeveloper = 1 << 17,
+	CertifiedModerator = 1 << 18,
```
Comment (vladfrangu) [inline payloads/v8/user.ts:101]: I know this is edging the scope a bit, but I think we should now add the description to each flag too...

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Comment by kyranet** [comment]
> It's funky with HypeSquad.. the name is one word, so it should be `Hypesquad` but their description says `HypeSquad`...

I think we should bring this to Discord's developers so we can make a PR fixing either the name of the description, then reflect the changes in this PR.

File: payloads/v8/user.ts (hunk 1 above)
Comment (almeidx) [reply payloads/v8/user.ts]: Done

**Comment by vladfrangu** [comment]
The website has it as `HypeSquad` too, see https://discord.com/hypesquad

This is more of a converting from UPPER_SNAKE_CASE to PascalCase problem. I suppose the correct form is `Hypesquad` for the enum member, even if the description says it otherwise

**Comment by iCrawl** [comment]
> The website has it as `HypeSquad` too, see https://discord.com/hypesquad

Yes about that.
image

**Review by vladfrangu — APPROVED** [review]

File: payloads/v8/user.ts [hunk 2]
```diff
@@ -85,20 +85,65 @@ export interface APIUser {
...
+	 */
+	HypesquadOnlineHouse1 = 1 << 6,
+	/**
+	 * House Brilliance Member
+	 */
+	HypesquadOnlineHouse2 = 1 << 7,
+	/**
+	 * House Balance Member
+	 */
+	HypesquadOnlineHouse3 = 1 << 8,
```
Comment (kyranet) [inline payloads/v8/user.ts:119]: Regardless of how the docs name those flags, the correct name is `HypeSquad`, as shown in https://discord.com/hypesquad
```suggestion
	/**
	 * House Bravery Member
	 */
	HypeSquadOnlineHouse1 = 1 << 6,
	/**
	 * House Brilliance Member
	 */
	HypeSquadOnlineHouse2 = 1 << 7,
	/**
	 * House Balance Member
	 */
	HypeSquadOnlineHouse3 = 1 << 8,
```

Comment (kyranet) [reply payloads/v8/user.ts:119]: Also apply those changes to v9 and generate Deno's types to match.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**